spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -4078,8 +4078,10 @@ def _handle_completions(self):
             cursor.setPosition(pos - 1, QTextCursor.MoveAnchor)
             cursor.select(QTextCursor.WordUnderCursor)
             prev_text = to_text_string(cursor.selectedText())
-            cursor.setPosition(pos + 1, QTextCursor.MoveAnchor)
-            if prev_text == '':
+            cursor.setPosition(pos - 1, QTextCursor.MoveAnchor)
+            cursor.setPosition(pos, QTextCursor.KeepAnchor)
+            prev_char = cursor.selectedText()
+            if prev_text == '' or prev_char in ('\u2029', ' ', '\t'):

```

Comment:
I think you need to use `u'\u2029'` here, for this to work in Python 2. We still support Python 2 an since our tests are failing only in that version, this could be the problem.

Reply:
Thank for the trick. I didn't even think to write something compatible with python2.